### ISSUE TITLE
Auth: Add more login methods

### ISSUE THREAD

**Issue by euclaise** [animenotifier/notify.moe#158]
Allow people to sign up and login using more than just Google/Facebook.  Ideally email, possibly OpenID or something (maybe GitHub even?).

**Comment by akyoto** [comment]
"Add more" is a little too vague for the issue title.
It's hard to mark an issue like this as completed.

Twitter and GitHub are possible.

The best is probably to create a separate issue for each provider so we can mark them as completed when the provider receives a login implementation.

E-Mail would require us to store password hashes which we are currently avoiding for security reasons.
It is not impossible but it requires a drastic architecture change just to account for passwords.

I am personally not interested in E-Mail login but I'd welcome any pull requests if someone wants to implement more login methods. [/auth](https://github.com/animenotifier/notify.moe/tree/go/auth) contains the authorization code.

**Comment by euclaise** [comment]
> "Add more" is a little too vague for the issue title.

Good point, although I'm not sure how to phrase it non-ambiguously, as "add alternative option(s) to google and FB" has the same issue

> E-Mail would require us to store password hashes which we are currently avoiding for security reasons.

It doesn't really have a large advantage to anyone but those who live under rocks (like me), so it makes sense to avoid it.  Yet, it's still always nice to have from a user-perspective

> I'd welcome any pull requests if someone wants to implement more login methods. /auth contains the authorization code.

I'll look into adding something when I can.

**Comment by akyoto** [comment]
- [ ] Twitter login #161 
- [ ] E-Mail login #163 

Now we can track completion for this issue :)

**Comment by ghost** [comment]
yeah.. I don't have a google, twitter or facebook account...
